components/Progress-Bar/progress-bar.js

```diff
@@ -0,0 +1,70 @@
+const msdsProgressBar = {
+  //Get the base number for the progress bar
+  parentElement: '',
+  baseNumber: 0,
+  currentStep: 0,
+  startFromOne: false,
+  init: function(parentElementId, baseNumber, customProgressionStep) {
+    this.parentElement = document.getElementById(parentElementId)
+
+    if (customProgressionStep > 0) {
+      this.currentStep = customProgressionStep
+    } else if (this.startFromOne == true) {
+      this.currentStep = 1
+    } else {
+      this.currentStep = 0
+    }
+    this.baseNumber = baseNumber
+    this.updateProgressbarDisplay()
+  },
+  updateProgressbarDisplay: function() {
+    //calculate the percentage width of the progress step and update the look and fill base on this
+    const updatedWidth = (this.currentStep * 100) / this.baseNumber
+
+    //update the progression bar
+    const progressBarProgressionElement = this.parentElement.querySelectorAll('.msds-progress-bar__progression-bar')
+    for (let i = 0; i < progressBarProgressionElement.length; i++) {
+      //update the width of the progression bar
```

Comment:
This comment could be deleted imo. line 29 tells me that you update the style.width with what ever is in updatedWidth

Reply:
same as above. :)